### PR TITLE
Kevinf/1905 uwp navigation

### PR DESCRIPTION
**Pull request by Bowman74**
## :sparkles: What kind of change does this PR introduce? (Bug fix, feature, docs update...)
Corrects navigation in UWP applications so that view models that are created as part of the navigation service are properly passed to the new view.

## :arrow_heading_down: What is the current behavior?
The a second copy of the view model is created and used by the view and does not receive any parameters passed to the navigation service.

## :new: What is the new behavior (if this is a feature change)?
The view model created by the navigation service is passed to and used by the view.

## :boom: Does this PR introduce a breaking change?
No.

## :memo: Links to relevant issues/docs
#1905 

## :thinking: Checklist before submitting

- [X] All projects build
- [X] Follows style guide lines ([code style guide](https://github.com/MvvmCross/MvvmCross#code-style-guidelines))
- [ ] Relevant documentation was updated ([docs style guide](https://www.mvvmcross.com/documentation/contribute/mvvmcross-docs-style-guide))
- [ ] Nuspec files were updated (when applicable)
- [ ] Rebased onto current develop

### CODE REVIEW

**Comment by martijn00** [comment]
Looks good! The only problem i see is that we now have a duplicate ChildViewModelCache. We could potentially move this to Core, but i'm not sure if that's the best option in the long run.

**Comment by Bowman74** [comment]
@martijn00 I was wondering the same thing. While it is only used in two places it is a bit questionable. If it ever gets used in a third it would become pretty clear it should go into Core. Let me know what you want me to do.

File: MvvmCross/Windows/Uwp/Views/MvxWindowsViewPresenter.cs [hunk 1]
```diff
@@ -29,11 +29,18 @@ public override void Show(MvxViewModelRequest request)
...
             try
             {
-                var requestTranslator = Mvx.Resolve<IMvxViewsContainer>();
-                var viewType = requestTranslator.GetViewType(request.ViewModelType);
-
-                var converter = Mvx.Resolve<IMvxNavigationSerializer>();
-                var requestText = converter.Serializer.SerializeObject(request);
+                var requestTranslator = Mvx.Resolve<IMvxWindowsViewModelRequestTranslator>();
+                string requestText = string.Empty;
+                if (request is MvxViewModelInstanceRequest)
```
Comment (Cheesebaron) [inline MvvmCross/Windows/Uwp/Views/MvxWindowsViewPresenter.cs]: can be written `if (request is MvxViewModelInstanceRequest instanceRequest)` and cast below removed.

**Comment by Cheesebaron** [comment]
Please rebase on develop and move MvxChildViewModelCache into core.

**Comment by Bowman74** [comment]
@Cheesebaron I have moved the interface, class and registration to the core project and re-pulled from develop into the branch.

**Comment by Cheesebaron** [comment]
I want you to rebase on develop, not just merge develop in again. Do you know how to do that? Are you perhaps using something like SourceTree, Kraken, Tower or similar to manage Git repositories or are you using command line?

**Comment by Bowman74** [comment]
@Cheesebaron I always use SourceTree, I spend way too much time in too many IDE's for my source control UI to change out from under me too. I assume you want to squash the commits into one, sure.

**Comment by Cheesebaron** [comment]
No, I don't want them squashed. I wan't them rebased to avoid all the Merge commit messages. If you go to SourceTree, stand on your branch (make sure you have latest develop checked out). Right click the develop branch. Click "Rebase current changes on top of develop". Force push your branch to your repo.

**Comment by Bowman74** [comment]
@Cheesebaron I don't think any of that worked at all for me.

**Comment by Cheesebaron** [comment]
Dunno, worked fine when I just did it locally. Rebased just fine. You don't want to do a pull before you push. That is why I asked you to force push.

**Comment by Bowman74** [comment]
@Cheesebaron Yeah, that was my mistake. Realized after I did it. I just saw the other requested fix with the C# 7 cast and setting in variable on same line, want me to do that too?

**Comment by Cheesebaron** [comment]
No it is OK. Will just do the rebase here and merge it.

**Comment by Cheesebaron** [comment]
This has been merged.

**Comment by Bowman74** [comment]
@Cheesebaron Thanks. Next time I'll get the rebase right.

**Comment by Cheesebaron** [comment]
Yeah, don't worry 😄 

Just to clear up. A rebase alters the history in a branch. So in your case, you did some changes, merged in develop a couple of times. Did some changes on top of that etc.

I wanted to get rid of the merge develop into branch commits as they are verbose. Doing a rebase, basically yanks your changes, gets latest develop and reapplies them.

You can also do a interactive rebase, where you can remove commits, change their messages and squash them.

Either way you are altering the history, so in order to push these changes, you will need to force push.

**Pull request closed without merging**